### ISSUE TITLE
Basic intro questions

### ISSUE THREAD

**Issue by k-damage** [rncm-prism/prism-samplernn#7]
Hello,

To preface, I am a beginner with using not only google colab but running neural networks and everything in general, but I will try my best to explain the problems I am running into/what i suspect the errors might be.

I have tried a few times to train with the suggest file type and sample rate (I have tried many differing lengths of audio files with no different results) and seem to only get a very high pitch frequency as a result in the predict folder and the generated test.wav file from after the training.

I am running the default settings on the colab and suspect I might need to edit some of the parameters along the way but I am a bit lost. I also don't seem to get any error codes or even warnings that seem noteworthy so I am not sure it is a problem with the script as much as the parameters/input file maybe?

Any help/feedback/references would be great!

**Comment by relativeflux** [comment]
Hi there,

The Colab notebook needs a bit of an update (the text is a little bit out of date)... Do you have the latest code? The train and predict folders are gone, they're replaced by a single, time-stamped folder. Grab the current version and upload to you Google Drive.

What are your training parameters? And also you generation parameters? And what is the size of your dataset?

**Comment by k-damage** [comment]
I have made a copy as of this morning so I believe everything would be up to date, but would there be anything I had to manually update within the colab notebook? 
I am also not sure what you are referring to when saying the train and predict folders are gone and replaced by a single, time -stamped folder. In my folder structure I see everything set up with the two folders accordingly.

"!python train.py \
  --data_dir ./chunks \
  --num_epochs 20 \
  --batch_size 64 \
  --max_checkpoints 2 \
  --checkpoint_every 500 \
  --output_file_dur 1 \
  --sample_rate 11025"

"!python generate.py \
  --output_path ./generated/default/test.wav \
  --checkpoint_path ./logdir/default/predict/model.ckpt-20 \
  --dur 1 \
  --sample_rate 16000 \
  --temperature 0.7 \
  --config_file ./default.config.json"

size of data set as in chunks or audio length? 123 chunks of 8 seconds each I believe to be around 16:30(note I know it is a smaller size but I was just testing as a beginner)

Thank you so much for the quick feedback!!!

**Comment by relativeflux** [comment]
Hi,

I guess if you downloaded the archive again it just overwrote the old folder, which would still contain `train` and `predict` subdirectories under `logdir`. But these are no longer needed. Checkpoints are now saved under timestamped subdirectories in this format:

`logdir/default/26.07.2020_20.48.51/model.ckpt-100`

The shortest interval at which a checkpoint can now be saved is 1 epoch (the `checkpoint_every` param now applies to epochs). Take a look at the list of updates now available at the top of the README on GitHub.

Actually (and apologies for this) I'd suggest copying the Colab notebook again, deletng your old prism-samplernn folder in your Drive and starting again. I have just updated the notebook, please let me know if wncounter any issues with it. Instead of downloading an archive of the code the notebook now allows you to clone the code using Git, which is inbuilt into Colab. This will allow you to just grab the code once, and then use `git pull` to update the code for any changes (see the notebook for further explanation).

Yes I meant size as in number of chunks. Given that the dataset will be truncated to the closest multiple of the batch size, if your batch size is 64 (as in the default) a dataset of 123 files will be truncated to the size of just one batch... So you'd be better off wither working with a larger dataset (I'd recommend that, 123 chunks is a bit small), or reducing you batch size, say to 32. To get a bigger dataset of chunks with your original audio you could always try reducing the duration of the chunks, say to 6 seconds.

Hope that helps.

**Comment by k-damage** [comment]
I will retry with these in mind immediately! thank you so much again!

**Comment by k-damage** [comment]
Seems to be working when taking everything into consideration and updating the parameters/colab respectfully!
I have a question about bitrate, since I am not too familiar with audio files in general this might be an obvious/noob question, but I was wondering if I put higher bitrate wav files for the training/generation would the quality be better? and would this in return make the training take longer etc because of file sizes?

**Comment by relativeflux** [comment]
Hi there,

Sorry for not getting back to you sooner. Glad you got the notebook working. Internally the system quantizes anything sent in to whatever you set for the q_levels. Which is 256 by default (8-bit). Testing hasn't revealed much difference when increasing this value, but it can certainly be slower. You can also change the quantization type from mu-law (which will always use 256 q_levels, no matter what you set for that param) to linear.

Chris

**Comment by mttjcksn** [comment]
Hi there, I hope you don't mind me hijacking this issue to ask a basic question.

What kind of training accuracy would you expect to see from a network that reproduces audio quite well?  I have a data set of about 2.5 hours of audio, which I have tried splitting into chunks of 1000, 2000, 4000, and 8000 seconds.  I've also tried up to 2048 dimensions and sequence lengths, lstm and gru.  I have tries lots of different settings for learning rate and learning rate reduction too, from between 0.005 and 0.0001. No matter what I try, It rarely gets past 50% and eventually gradually reduces in accuracy. How many epochs this takes obviously depends on the configuration, but I can hit 50% from anywhere between 2 and 10 epochs.  Am I missing something obvious? Training audio is normalised. 

Many thanks for your help and for this repo!

**Comment by relativeflux** [comment]
@mttjcksn Hi, thanks for the question. When you say chunks of 8000 seconds, do you mean miliseconds? That's what the `chunk_length` parameter for the chunking script means. Also what is your batch size?

Simlarly to the Dadabots implementation, our best results have come from LSTM with skip connections plus linear quantization. There is a new branch validation which I am going to merge into master later today. This prints validation loss and accuracy, and also fixes a few bugs (for example, linear quantization was broken). These provide good information on the progress of training, particularly with respect to over/underfitting.

Another very important factor is the size of the dataset, although with 2.5 hrs of input audio and 8 second chunks that should give you a decent number of chunks. > 1000 chunks works well. You can also use the overlap param to the chunking script to get a larger number of chunks, in fact an overlap of 7 seconds with an 8 second chunk size seems fine.

**Comment by mttjcksn** [comment]
Thanks for a fast reply @relativeflux !

My batch size is usually around 64.. I take whatever the nearest divisor of the number of chunks is nearest to 64.  I always make sure I have at least a couple of thousand chunks.

I haven't tried skip_conn or linear quantization yet.

Perhaps I will try again tomorrow with a bigger batch size, bigger overlap between samples, skip_conn, and linear quantization (Provided you manage to merge today). I'll let you know how I get on.

Many thanks
